File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.c
int totalNumbers(int* digits, int digitsSize) {
    int freq[10] = {0};

    for (int i = 0; i < digitsSize; i++) {
        freq[digits[i]]++;
    }

    int ans = 0;

    for (int num = 100; num <= 999; num++) {

        // Must be even
        if (num % 2 != 0)
            continue;

        int ones = num % 10;
        int tens = (num / 10) % 10;
        int hundreds = num / 100;

        int need[10] = {0};

        need[hundreds]++;
        need[tens]++;
        need[ones]++;

        int possible = 1;

        for (int d = 0; d <= 9; d++) {
            if (need[d] > freq[d]) {
                possible = 0;
                break;
            }
        }

        if (possible)
            ans++;
    }

    return ans;
}

File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.py
class Solution:
    def totalNumbers(self, digits):
        freq = [0] * 10

        for d in digits:
            freq[d] += 1

        ans = 0

        for num in range(100, 1000):

            # Must be even
            if num % 2 != 0:
                continue

            ones = num % 10
            tens = (num // 10) % 10
            hundreds = num // 100

            need = [0] * 10

            need[hundreds] += 1
            need[tens] += 1
            need[ones] += 1

            possible = True

            for d in range(10):
                if need[d] > freq[d]:
                    possible = False
                    break

            if possible:
                ans += 1

        return ans